stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -1311,6 +1323,62 @@
       (ok { stacker: stacker,
             unlock-burn-height: new-unlock-ht }))))
 
+;; This function allows a node to register a new 'current-signer-set' during the prepare phase
+(define-private (update-current-signer-set (new-signer-set (list 4096 {signer: (buff 33)})))
+    (let 
+        (
+            (next-reward-cycle (+ u1 current-pox-reward-cycle))
+            (next-reward-cycle-burn-height (reward-cycle-to-burn-height next-reward-cycle))
+            (next-signer-set-map (map-get? reward-cycle-signing-key-ids next-reward-cycle))
```

Comment:
This definitely doesn't type-check -- the key type for `reward-cycle-signing-key-ids` is `{ cycle: uint, signer: (buff 33) }` not `uint`.